src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php

```diff
@@ -127,11 +135,28 @@ public function delete(NodeEvent $event)
     private function hasOfflineParents(NodeTranslation $nodeTranslation)
     {
         $lang = $nodeTranslation->getLang();
-        foreach ($nodeTranslation->getNode()->getParents() as $node) {
-            $nodeNT = $node->getNodeTranslation($lang, true);
-            if ($nodeNT && !$nodeNT->isOnline() && !$nodeNT instanceof StructureNode) {
-                return true;
+        $node = $nodeTranslation->getNode();
+        if (null !== $this->em) {
+            $em = $this->em;
+        } elseif (null !== $this->container) {
+            $em = $this->container->get('doctrine.orm.entity_manager.abstract');
+        } else {
+            /** Revert to bugged behaviour to avoid BC breaking */
+            return false;
```

Comment:
This will return a different value as it would be the case in the previous (buggy) behavior? Or am I wrong?